### PR TITLE
fix: errors fixed in Supplier Scorecard

### LINKED ISSUE #16255: The child tables of Supplier Scorecard does not have a preconfigured fetch
I was trying to create a Supplier Scorecard, but there are some hidden fields that are mandatory and does not have the correspondent "fetch" option presetted. Although the main problem is with the "variables" child table, the other child tables are missing some "fetch" options too (all of them really).
And I see that is_custom, in Supplier Scorecard Scoring Variable, is mandatory, is that intended?

ERPNext: v10.1.75 (master)
Frappe Framework: v10.1.68 (master)

Here are some gifs:
Creation and Supplier Scorecard Scoring Standing:
First child table

Supplier Scorecard Scoring Criteria:
Second child table

Supplier Scorecard Scoring Variable:
Third child table

Missing fields error:
Final error

### PR DESCRIPTION
**Pull request by Don-Leopardo**
Resolves #16255  
An extra error in the creation of Supplier Scorecard Period:
captura de pantalla 2018-12-28 a la s 10 58 33

And some small bug in javascript and front end (related to fetches and console logs).

### CODE REVIEW

**Comment by federicocalvo** [comment]
Hi @sagarvora, any news for this PR? This functionality never worked, that's why it's necessary to merge the PR so you can start using it at least.

**Comment by sagarvora** [comment]
@federicocalvo I'll test it manually, thanks for informing.

File: erpnext/buying/doctype/supplier_scorecard/supplier_scorecard.js [hunk 1]
```diff
@@ -13,24 +13,20 @@ frappe.ui.form.on("Supplier Scorecard", {
 	},
 	onload: function(frm) {
 		if (frm.doc.__unsaved == 1)	{
+			frm.doc.variables = [];
```
Comment (sagarvora) [inline erpnext/buying/doctype/supplier_scorecard/supplier_scorecard.js]: why do we need this in the first place?

**Comment by sagarvora** [comment]
Closing for now.

**Pull request closed without merging**